grudge/dof_desc.py

```diff
@@ -241,20 +286,45 @@ def fmt(s):
 
         return "DOFDesc({}, {})".format(
                 fmt(self.domain_tag),
-                fmt(self.quadrature_tag))
+                fmt(self.discretization_tag))
 
 
 DD_SCALAR = DOFDesc(DTAG_SCALAR, None)
 
 DD_VOLUME = DOFDesc(DTAG_VOLUME_ALL, None)
 
-DD_VOLUME_MODAL = DOFDesc(DTAG_VOLUME_ALL, QTAG_MODAL)
+DD_VOLUME_MODAL = DOFDesc(DTAG_VOLUME_ALL, DISCR_TAG_MODAL)
 
 
 def as_dofdesc(dd):
     if isinstance(dd, DOFDesc):
         return dd
-    return DOFDesc(dd, quadrature_tag=None)
+    return DOFDesc(dd, discretization_tag=None)
+
+# }}}
+
+
+# {{{ Deprecated tags
+
+_deprecated_names = {"QTAG_NONE": "DISCR_TAG_BASE",
+                     "QTAG_MODAL": "DISCR_TAG_MODAL"}
+
+
+def __getattr__(name):
+    if name in _deprecated_names:
+        warn(f"{name} is deprecated and will be dropped "
+             f"in version 2022.x. Use {_deprecated_names[name]} instead.",
+             DeprecationWarning, stacklevel=2)
+        return globals()[f"{_deprecated_names[name]}"]
+
+    raise AttributeError(f"module {__name__} has no attribute {name}")
+
+
+import sys
```

Comment:
Move the import to the top?

Reply:
Done: https://github.com/inducer/grudge/pull/83/commits/f0b8bdf2bf05f13c019530025a3039afc045f6a2